frappe/database/database.py

```diff
@@ -960,16 +963,24 @@ def begin(self, *, read_only=False):
 		read_only = read_only or frappe.flags.read_only
 		mode = "READ ONLY" if read_only else ""
 		self.sql(f"START TRANSACTION {mode}")
+		self.in_transaction = True
 
-	def commit(self):
+	def commit(self, new_transaction: bool = True):
 		"""Commit current transaction. Calls SQL `COMMIT`."""
+		if not self.in_transaction:
+			self.logger.warn("Not in transaction, ignoring commit")
+			return
```

Comment:
IMO we should never change behaviour of db.commit because even if we haven't made writes or in transaction, there are valid usage of committing/rolling back.

Reply:
Reverted this, switched to `frappe.db.sql_ddl()` where required to avoid implicit commit issue.